sbt-plugin/src/main/scala/ch/epfl/scala/AnalyzeDependencyGraph.scala

```diff
@@ -0,0 +1,216 @@
+package ch.epfl.scala
+
+import java.nio.file.Paths
+
+import scala.Console
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.sys.process._
+import scala.util.Failure
+import scala.util.Properties
+import scala.util.Success
+import scala.util.Try
+
+import ch.epfl.scala.GithubDependencyGraphPlugin.autoImport._
+import ch.epfl.scala.githubapi._
+import gigahorse.FullResponse
+import gigahorse.HttpClient
+import gigahorse.support.asynchttpclient.Gigahorse
+import sbt._
+import sbt.internal.util.complete._
+import sjsonnew.shaded.scalajson.ast.unsafe.JArray
+import sjsonnew.shaded.scalajson.ast.unsafe.JField
+import sjsonnew.shaded.scalajson.ast.unsafe.JObject
+import sjsonnew.shaded.scalajson.ast.unsafe.JString
+import sjsonnew.support.scalajson.unsafe.{Parser => JsonParser}
+
+object AnalyzeDependencyGraph {
+
+  val help =
+    "download and display CVEs alerts from Github, and analyze them against dependencies (use hub or gh local config or GIT_TOKEN env var to authenticate, requires githubGenerateSnapshot)"
+
+  case class AnalysisParams(repository: Option[String])
+
+  val AnalyzeDependencies = "githubAnalyzeDependencies"
+  private val AnalyzeDependenciesUsage =
+    s"""$AnalyzeDependencies [pattern]"""
+  private val AnalyzeDependenciesDetail = s"""Analyze the dependencies based on a search pattern:
+  $help
+  """
+
+  val commands: Seq[Command] = Seq(
+    Command(AnalyzeDependencies, (AnalyzeDependenciesUsage, AnalyzeDependenciesDetail), AnalyzeDependenciesDetail)(
+      parser
+    )(analyzeDependencies)
+  )
+
+  private def parser(state: State): Parser[AnalysisParams] =
+    Parsers.any.*.map { raw =>
+      raw.mkString.trim.split(" ").toSeq match {
+        case Seq("") | Nil => AnalysisParams(None)
+        case Seq(arg)      => AnalysisParams(Some(arg))
+      }
+    }.failOnException
+
+  private def analyzeDependencies(state: State, params: AnalysisParams): State =
+    (for {
+      repo <- params.repository.orElse(getGitHubRepo)
+      vulnerabilities <- downloadAlerts(state, repo) match {
+        case Success(v) => Some(v)
+        case Failure(e) =>
+          state.log.error(s"Failed to download alerts: ${e.getMessage}")
+          None
+      }
+    } yield analyzeCves(state, vulnerabilities)).getOrElse(state)
```

Comment:
The state is always untouched, and so you could write:
```suggestion
    } analyzeCves(state, vulnerabilities)
  state
```